// svix/dist/api/event_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventType = void 0;
const openapi_1 = require("../openapi");
class EventType {
    constructor(config) {
        this.api = new openapi_1.EventTypeApi(config);
    }
    list(options) {
        var _a;
        return this.api.v1EventTypeList(Object.assign(Object.assign({}, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined }));
    }
    create(eventTypeIn, options) {
        return this.api.v1EventTypeCreate(Object.assign({ eventTypeIn }, options));
    }
    importOpenapi(eventTypeImportOpenApiIn, options) {
        return this.api.v1EventTypeImportOpenapi(Object.assign({ eventTypeImportOpenApiIn }, options));
    }
    get(eventTypeName) {
        return this.api.v1EventTypeGet({
            eventTypeName,
        });
    }
    update(eventTypeName, eventTypeUpdate) {
        return this.api.v1EventTypeUpdate({
            eventTypeName,
            eventTypeUpdate,
        });
    }
    delete(eventTypeName) {
        return this.api.v1EventTypeDelete({
            eventTypeName,
        });
    }
    patch(eventTypeName, eventTypePatch) {
        return this.api.v1EventTypePatch({
            eventTypeName,
            eventTypePatch,
        });
    }
}
exports.EventType = EventType;
