// svix/dist/api/integration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Integration = void 0;
const openapi_1 = require("../openapi");
class Integration {
    constructor(config) {
        this.api = new openapi_1.IntegrationApi(config);
    }
    list(appId, options) {
        var _a;
        return this.api.v1IntegrationList(Object.assign(Object.assign({ appId }, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined }));
    }
    create(appId, integrationIn, options) {
        return this.api.v1IntegrationCreate(Object.assign({ appId,
            integrationIn }, options));
    }
    get(appId, integId) {
        return this.api.v1IntegrationGet({
            appId,
            integId,
        });
    }
    update(appId, integId, integrationUpdate) {
        return this.api.v1IntegrationUpdate({
            appId,
            integId,
            integrationUpdate,
        });
    }
    delete(appId, integId) {
        return this.api.v1IntegrationDelete({
            appId,
            integId,
        });
    }
    getKey(appId, integId) {
        return this.api.v1IntegrationGetKey({
            integId,
            appId,
        });
    }
    rotateKey(appId, integId, options) {
        return this.api.v1IntegrationRotateKey(Object.assign({ appId,
            integId }, options));
    }
}
exports.Integration = Integration;
