// svix/dist/api/message_attempt.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageAttempt = void 0;
const openapi_1 = require("../openapi");
class MessageAttempt {
    constructor(config) {
        this.api = new openapi_1.MessageAttemptApi(config);
    }
    list(appId, msgId, options) {
        return this.listByMsg(appId, msgId, options);
    }
    listByEndpoint(appId, endpointId, options) {
        var _a, _b, _c;
        return this.api.v1MessageAttemptListByEndpoint(Object.assign(Object.assign({ appId,
            endpointId }, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined, before: (_b = options === null || options === void 0 ? void 0 : options.before) !== null && _b !== void 0 ? _b : undefined, after: (_c = options === null || options === void 0 ? void 0 : options.after) !== null && _c !== void 0 ? _c : undefined }));
    }
    listByMsg(appId, msgId, options) {
        var _a, _b, _c;
        return this.api.v1MessageAttemptListByMsg(Object.assign(Object.assign({ appId,
            msgId }, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined, before: (_b = options === null || options === void 0 ? void 0 : options.before) !== null && _b !== void 0 ? _b : undefined, after: (_c = options === null || options === void 0 ? void 0 : options.after) !== null && _c !== void 0 ? _c : undefined }));
    }
    listAttemptedMessages(appId, endpointId, options) {
        var _a, _b, _c;
        return this.api.v1MessageAttemptListAttemptedMessages(Object.assign(Object.assign({ appId,
            endpointId }, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined, before: (_b = options === null || options === void 0 ? void 0 : options.before) !== null && _b !== void 0 ? _b : undefined, after: (_c = options === null || options === void 0 ? void 0 : options.after) !== null && _c !== void 0 ? _c : undefined }));
    }
    get(appId, msgId, attemptId) {
        return this.api.v1MessageAttemptGet({
            appId,
            msgId,
            attemptId,
        });
    }
    expungeContent(appId, msgId, attemptId) {
        return this.api.v1MessageAttemptExpungeContent({
            appId,
            msgId,
            attemptId,
        });
    }
    listAttemptedDestinations(appId, msgId, options) {
        var _a;
        return this.api.v1MessageAttemptListAttemptedDestinations(Object.assign(Object.assign({ appId,
            msgId }, options), { iterator: (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined }));
    }
    resend(appId, msgId, endpointId, options) {
        return this.api.v1MessageAttemptResend(Object.assign({ appId,
            msgId,
            endpointId }, options));
    }
}
exports.MessageAttempt = MessageAttempt;
