// svix/dist/api/op_webhook_endpoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationalWebhookEndpoint = void 0;
const openapi_1 = require("../openapi");
class OperationalWebhookEndpoint {
    constructor(config) {
        this.api = new openapi_1.WebhookEndpointApi(config);
    }
    list(options) {
        var _a;
        const iterator = (_a = options === null || options === void 0 ? void 0 : options.iterator) !== null && _a !== void 0 ? _a : undefined;
        return this.api.listOperationalWebhookEndpoints(Object.assign(Object.assign({}, options), { iterator }));
    }
    create(endpointIn, options) {
        return this.api.createOperationalWebhookEndpoint(Object.assign({ operationalWebhookEndpointIn: endpointIn }, options));
    }
    get(endpointId) {
        return this.api.getOperationalWebhookEndpoint({ endpointId });
    }
    update(endpointId, endpointUpdate) {
        return this.api.updateOperationalWebhookEndpoint({
            endpointId,
            operationalWebhookEndpointUpdate: endpointUpdate,
        });
    }
    delete(endpointId) {
        return this.api.deleteOperationalWebhookEndpoint({ endpointId });
    }
    getSecret(endpointId) {
        return this.api.getOperationalWebhookEndpointSecret({ endpointId });
    }
    rotateSecret(endpointId, endpointSecretIn, options) {
        return this.api.rotateOperationalWebhookEndpointSecret(Object.assign({ endpointId, operationalWebhookEndpointSecretIn: endpointSecretIn }, options));
    }
}
exports.OperationalWebhookEndpoint = OperationalWebhookEndpoint;
