// svix/src/openapi/models/RotatePollerTokenIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class RotatePollerTokenIn {
    /**
    * How long the token will be valid for, in seconds. Can be up to 31,536,000 seconds (1 year).
    */
    'expiry'?: number | null;
    /**
    * Updates the previous token\'s expiration, in seconds. If set to 0, the old token will immediately be revoked. Must be between 0 and 86,400 seconds (1 day).  Defaults to 300 seconds (5 minutes).
    */
    'oldTokenExpiry'?: number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "expiry",
            "baseName": "expiry",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "oldTokenExpiry",
            "baseName": "oldTokenExpiry",
            "type": "number",
            "format": "int64"
        }    ];

    static getAttributeTypeMap() {
        return RotatePollerTokenIn.attributeTypeMap;
    }

    public constructor() {
    }
}
